src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -1223,6 +1227,30 @@ protected override void WriteReferenceCaptureInnards(CodeRenderingContext contex
         }
     }
 
+    public override void WriteRenderMode(CodeRenderingContext context, RenderModeIntermediateNode node)
+    {
+        // Looks like:
+        // global::Microsoft.AspNetCore.Components.IComponentRenderMode __renderMode = expression;
```

Comment:
Do we have a test where there are two components with `@rendermode` attributes? Want to see that these decls are in different scopes / methods so they don't conflict. Basically 

```html
<input @rendermode="Server" /> 
<input @rendermode="Server" /> 
```